File: evals/fixtures/cowork-php-laravel/app/app/Services/BookingService.php
<?php

namespace App\Services;

use App\Models\Store;

class BookingService
{
    public static function createBooking(array $data): array
    {
        $room = Store::findRoom($data['room_id'] ?? '');
        if (!$room) {
            throw new \InvalidArgumentException("Room not found: " . ($data['room_id'] ?? ''));
        }

        // Find existing member by email or create a new one
        $member = null;
        foreach (Store::getMembers() as $m) {
            if ($m['email'] === ($data['member_email'] ?? '')) {
                $member = $m;
                break;
            }
        }

        if (!$member) {
            $member = [
                'id' => Store::generateId(),
                'name' => $data['member_name'] ?? '',
                'email' => $data['member_email'] ?? '',
                'company' => $data['member_company'] ?? '',
            ];
            Store::addMember($member);
        }

        $booking = [
            'id' => Store::generateId(),
            'member_id' => $member['id'],
            'room_id' => $data['room_id'],
            'start_time' => $data['start_time'],
            'end_time' => $data['end_time'],
            'status' => 'active',
        ];

        Store::addBooking($booking);

        return $booking;
    }

    public static function updateBooking(string $id, array $data): array
    {
        $result = Store::updateBookingById($id, function (array $booking) use ($data) {
            if (isset($data['start_time'])) {
                $booking['start_time'] = $data['start_time'];
            }
            if (isset($data['end_time'])) {
                $booking['end_time'] = $data['end_time'];
            }
            return $booking;
        });

        if (!$result) {
            throw new \InvalidArgumentException("Booking not found: {$id}");
        }

        return $result;
    }

    public static function cancelBooking(string $id): array
    {
        $result = Store::updateBookingById($id, function (array $booking) {
            $booking['status'] = 'cancelled';
            return $booking;
        });

        if (!$result) {
            throw new \InvalidArgumentException("Booking not found: {$id}");
        }

        return $result;
    }

    public static function getBooking(string $id): array
    {
        $booking = Store::findBooking($id);

        if (!$booking) {
            throw new \InvalidArgumentException("Booking not found: {$id}");
        }

        return $booking;
    }
}

File: evals/fixtures/cowork-rails-rb/app/app/services/booking_service.rb
class BookingService
  class << self
    def create_booking(data)
      room = Store.find_room(data[:room_id])
      raise "Room not found: #{data[:room_id]}" unless room

      # Find existing member by email or create a new one
      member = Store.members.find { |m| m[:email] == data[:member_email] }
      unless member
        member = {
          id: Store.generate_id,
          name: data[:member_name],
          email: data[:member_email],
          company: data[:member_company]
        }
        Store.members << member
      end

      booking = {
        id: Store.generate_id,
        member_id: member[:id],
        room_id: data[:room_id],
        start_time: data[:start_time],
        end_time: data[:end_time],
        status: "active"
      }

      Store.bookings << booking
      booking
    end

    def update_booking(id, data)
      booking = Store.bookings.find { |b| b[:id] == id }
      raise "Booking not found: #{id}" unless booking

      booking[:start_time] = data[:start_time] if data[:start_time].present?
      booking[:end_time] = data[:end_time] if data[:end_time].present?

      booking
    end

    def cancel_booking(id)
      booking = Store.bookings.find { |b| b[:id] == id }
      raise "Booking not found: #{id}" unless booking

      booking[:status] = "cancelled"
      booking
    end

    def get_booking(id)
      booking = Store.bookings.find { |b| b[:id] == id }
      raise "Booking not found: #{id}" unless booking

      booking
    end
  end
end
